### ISSUE TITLE
Add Containers.Image resource

### ISSUE THREAD

**Issue by rbino** [edgehog-device-manager/edgehog#611]
For now, let's keep this simple and just have this single resource. We can decide to normalize it (breaking out Registry and Repo) with following tasks. Note that tag is still broken out.

Attributes:
```
id (uuid primary key) (example: 62902fc3-2de9-48dc-a6cb-5d84aa00b543)
reference (string, unique, non-null) (examples: europe-west4-docker.pkg.dev/rbino-personal/my-private-repo/hello-world, postgres)
credentials (belongs_to, references Containers.ImageCredentials)
```

Let's not add any GraphQL queries for now and just add them as we go to satisfy the frontend needs.

**Comment by noaccOS** [comment]
After some discussion, we've decided for a single resource `Image` instead:
- As multiple identifiers can refer to the same image (eg: `postgres` and `docker.io/postgres`), we could potentially have multiple images referring to the same actual image
- For this reason we don't actually have any advantage in normalizing this table
- This would be fixed by parsing the full string beforehand, but to do that we first need to better understand the concepts from the [distribution spec](https://github.com/opencontainers/distribution-spec/blob/main/spec.md)

**Comment by rbino** [comment]
Closed by #616
